pkg/data/result_test.go

```diff
@@ -0,0 +1,110 @@
+package data
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/testground/testground/pkg/runner"
+	"github.com/testground/testground/pkg/task"
+)
+
+func TestDecodeResult(t *testing.T) {
+	result1 := &runner.Result{
+		Outcome:  task.OutcomeUnknown,
+		Outcomes: make(map[string]*runner.GroupOutcome),
+		Journal: &runner.Journal{
+			Events:       make(map[string]string),
+			PodsStatuses: make(map[string]struct{}),
+		},
+	}
+	r1 := DecodeRunnerResult(result1)
+	assert.NotNil(t, r1)
+
+	result2 := [2]string{"artifact", "artifact2"}
+	r2 := DecodeRunnerResult(result2)
+	assert.NotNil(t, r2)
+}
+
+func TestDecodeTaskOutcome(t *testing.T) {
```

Comment:
I'd rather split this large test into several smaller unit tests, each one testing a specific outcome (and the decoding of the outcome).
E.g. `TestDecodeGenericUnknownOutcome`, `TestDecodeGenericSuccessOutcome`, etc.

Reply:
Fixed!